extensions/arc/src/models/resourceModel.ts

```diff
@@ -23,5 +26,61 @@ export abstract class ResourceModel {
 		this._onRegistrationUpdated.fire(this._registration);
 	}
 
+	/**
+ * Loads the saved connection profile associated with this model. Will prompt for one if
+ * we don't have one or can't find it (it was deleted)
+ */
+	protected async getConnectionProfile(externalEndpoint: string | undefined, providerName: string, userName: string | undefined): Promise<void> {
+		const ipAndPort = parseIpAndPort(externalEndpoint || '');
+		let connectionProfile: azdata.IConnectionProfile | undefined = {
+			serverName: `${ipAndPort.ip},${ipAndPort.port}`,
+			databaseName: '',
+			authenticationType: 'SqlLogin',
+			providerName: providerName,
+			connectionName: '',
+			userName: userName || '',
+			password: '',
+			savePassword: true,
+			groupFullName: undefined,
+			saveProfile: true,
+			id: '',
+			groupId: undefined,
+			options: {
+				host: `${ipAndPort.ip}`,
```

Comment:
Host is a PG only setting it looks like.

In general I think having each implementation provide their own version of this makes sense so that we can ensure only the necessary values are being set correctly. Maybe something like `createConnectionProfile` or something that this calls.